CMakeLists.txt

```diff
@@ -1,15 +1,32 @@
-cmake_minimum_required(VERSION 3.0.0)
+# CMakeLists.txt
+# Cubos Demo project root build configuration
+
+cmake_minimum_required(VERSION 3.9.0)
 project(cubos-demo VERSION 0.1.0)
 
-set(DEMO_SOURCE
-    "src/main.cpp"
-)
+# Include CubinhosGenerate
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/lib/cubos/cmake")
+include(CubinhosGenerate)
 
+# Build demo executabe
+set(DEMO_SOURCE "src/main.cpp")
 add_executable(demo ${DEMO_SOURCE})
 
 # Dependencies
 add_subdirectory(lib/cubos)
-target_link_libraries(demo cubos)
+target_link_libraries(demo cubos-core cubos-engine)
+target_include_directories(demo PUBLIC
+                          "${CMAKE_CURRENT_BINARY_DIR}"
+                          "lib/cubos"
+                          )
+set_property(TARGET demo PROPERTY CXX_STANDARD 20)
+target_compile_features(demo PUBLIC cxx_std_20)
```

Comment:
Does it compile without these 3 lines? It should because in CMake `target_link_libraries` 'imports' all public configuration of the targets, so this should be automatic